tools/sil-llvm-gen/swift_lto_main.cpp

```diff
@@ -0,0 +1,83 @@
+//===--- swift_lto_main.cpp -----------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// This is a tool for reading sib files and running LTO passes upon them.
+///
+//===----------------------------------------------------------------------===//
+
+#include "swift/Basic/LLVMInitialize.h"
+#include "swift/Frontend/Frontend.h"
+#include "swift/LTO/LTO.h"
+#include "swift/Strings.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include <cstdio>
+using namespace swift;
+
+static llvm::cl::list<std::string>
+    InputFilenames(llvm::cl::Positional, llvm::cl::desc("[input files...]"),
+                   llvm::cl::ZeroOrMore);
+
+void anchorForGetMainExecutable();
+
+int swift_lto_main(int argc, char **argv) {
+  PROGRAM_START(argc, argv);
+  INITIALIZE_LLVM();
+
+  llvm::cl::ParseCommandLineOptions(argc, argv, "Swift LTO Tool\n");
+
+  CompilerInvocation Invocation;
+
+  Invocation.setMainExecutablePath(llvm::sys::fs::getMainExecutable(
+      argv[0], reinterpret_cast<void *>(&anchorForGetMainExecutable)));
+
+  auto SearchPathOpts = Invocation.getSearchPathOptions();
+  llvm::SmallVector<llvm::StringRef, 4> RuntimeLibraryPaths;
+  llvm::SmallVector<llvm::StringRef, 4> RuntimeLibraryImportPaths;
+  RuntimeLibraryPaths.insert(RuntimeLibraryPaths.begin(),
+                             SearchPathOpts.RuntimeLibraryPaths.begin(),
+                             SearchPathOpts.RuntimeLibraryPaths.end());
+  RuntimeLibraryImportPaths.insert(
+      RuntimeLibraryImportPaths.begin(),
+      SearchPathOpts.RuntimeLibraryImportPaths.begin(),
+      SearchPathOpts.RuntimeLibraryImportPaths.end());
+  lto::LTOPipeline Pipeline(RuntimeLibraryPaths, RuntimeLibraryImportPaths,
+                            SearchPathOpts.RuntimeResourcePath);
+
+  for (auto InputFilename : InputFilenames) {
+    // Load the input file.
+    auto FileBufOrErr = llvm::MemoryBuffer::getFileOrSTDIN(InputFilename);
+    if (!FileBufOrErr) {
+      fprintf(stderr, "Error! Failed to open file: %s\n",
+              InputFilename.c_str());
+      exit(-1);
+    }
+
+    if (Pipeline.addModule(std::move(FileBufOrErr.get()))) {
+      fprintf(stderr, "Error! Failed to load serialized module: %s\n",
+              InputFilename.c_str());
+      exit(-1);
+    }
+  }
+
+  Pipeline.emitLLVMModules([&](StringRef ModuleName) {
+    std::error_code EC;
+    std::unique_ptr<llvm::raw_ostream> RawOS(
+        new llvm::raw_fd_ostream(ModuleName.str() + ".bc", EC));
```

Comment:
Can you get away with `std::make_unqiue` instead of the `new llvm::raw_fd_ostream`?